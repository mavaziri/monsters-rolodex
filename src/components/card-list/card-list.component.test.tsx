import { render, screen } from '@testing-library/react';

import CardList from './card-list.component';

const mockMonsters = [
  {
    id: '1',
    name: 'Monster One',
    email: 'monster1@example.com',
  },
  {
    id: '2',
    name: 'Monster Two',
    email: 'monster2@example.com',
  },
  {
    id: '3',
    name: 'Monster Three',
    email: 'monster3@example.com',
  },
];

test('renders Card List component with monsters details', () => {
  const { container } = render(<CardList monsters={mockMonsters} />);
  expect(container).toMatchSnapshot();
});
