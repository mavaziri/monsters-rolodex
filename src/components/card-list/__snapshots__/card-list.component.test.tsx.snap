// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card List component with monsters details 1`] = `
<div>
  <div
    class="card-list"
  >
    <div
      class="card-container"
    >
      <img
        alt="monster"
        src="https://robohash.org/1?set=set2&size=180x180"
      />
      <h2>
         
        Monster One
         
      </h2>
      <p>
        monster1@example.com
      </p>
    </div>
    <div
      class="card-container"
    >
      <img
        alt="monster"
        src="https://robohash.org/2?set=set2&size=180x180"
      />
      <h2>
         
        Monster Two
         
      </h2>
      <p>
        monster2@example.com
      </p>
    </div>
    <div
      class="card-container"
    >
      <img
        alt="monster"
        src="https://robohash.org/3?set=set2&size=180x180"
      />
      <h2>
         
        Monster Three
         
      </h2>
      <p>
        monster3@example.com
      </p>
    </div>
  </div>
</div>
`;
